youtube_dl/extractor/mwave.py

```diff
@@ -28,6 +28,14 @@ class MwaveIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
+        # Check that language (if given) is valid
+        languages = ['en/', 'jp/', 'cn/', 'tw/', 'kr/', 'es/']
+        start = url.index('mwave.interest.me/') + len('mwave.interest.me/')
+        end = url.index('mnettv/', start)
+        if url[start:end] and not url[start:end] in languages:
+            raise Exception('[mwave] invalid language detected: {}'.format(url[start:end]))
```

Comment:
This should be detected in `_VALID_URL` if reasonable. But there is also no sense checking it all since new languages may appear in future. There is also a code duplicate. `{}` won't work in python 2.6. For extractor errors `ExtractorError` should be used.